amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/rules/FileAsRecordRuleTests.java

```diff
@@ -0,0 +1,242 @@
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+    }
+    /**
+     * Given a record folder location has been selected for a "Declare and File as Record" action within a rule
+     * And I have permissions and capabilities to file to that record folder
+     * When I trigger the rule
+     * Then the file is filed directly to the selected record folder from the file plan
+     */
+
+    @Test
+    public void triggerDeclareToRecordFolderRuleAsUserWithPermissions()
+    {
+        STEP("Create as rmManager a new file into the folderWithRule in order to trigger the rule");
+        FileModel testFile = dataContent.usingUser(rmManager).usingResource(testFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+    }
+
+    /**
+     * Given a record folder location has been selected for a "Declare and File as Record" action within a rule
+     * And I don't have permissions and capabilities to file to that record folder
+     * When I trigger the rule
+     * Then the file is not declared as record
+     */
+
+    @Test
+    public void triggerDeclareToRecordFolderRuleAsUserWithoutPermissions()
+    {
+        STEP("Create as nonRMuser a new file into the folderWithRule in order to trigger the rule");
+        FileModel testFile = dataContent.usingUser(nonRMUser).usingResource(testFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+    }
+
+    /**
+     * Given I have not selected a record folder location
+     * When the rule is triggered
+     * Then the file is declared as record to the Unfiled Records folder
+     */
+    @Test
+    public void triggerDeclareToUnfiledRuleAsNonRMUser()
+    {
+        STEP("Create a collaboration folder with a rule set to declare and file as record without a record folder location");
+
+        RecordCategory recordCategory = new RecordCategory().builder()
+            .id(category_manager.getId()).build();
+        RecordCategoryChild folderWithRule = createFolder(recordCategory.getId(), getRandomName("recordFolder"));
+        RuleDefinition ruleDefinition = RuleDefinition.createNewRule().title("name").description("description")
+            .applyToChildren(true)
+            .actions(Collections.singletonList(ActionsOnRule.DECLARE_AS_RECORD.getActionValue()));
+        rulesAPI.createRule(getAdminUser().getUsername(), getAdminUser().getPassword(), NODE_PREFIX + folderWithRule.getId(), ruleDefinition);
+
+        STEP("Create as nonRMuser a new file into the previous folder in order to trigger the rule");
+        inPlaceRecord = dataContent.usingUser(nonRMUser).usingResource(testFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
```

Comment:
There is no checks done to make sure the file is present in unfiled records. Need to complete the test case.